lldb/tools/lldb-dap/lldb-dap.cpp

```diff
@@ -4028,6 +4046,161 @@ void request_disassemble(const llvm::json::Object &request) {
   response.try_emplace("body", std::move(body));
   g_dap.SendJSON(llvm::json::Value(std::move(response)));
 }
+
+// "ReadMemoryRequest": {
+//   "allOf": [ { "$ref": "#/definitions/Request" }, {
+//     "type": "object",
+//     "description": "Reads bytes from memory at the provided location. Clients
+//                     should only call this request if the corresponding
+//                     capability `supportsReadMemoryRequest` is true.",
+//     "properties": {
+//       "command": {
+//         "type": "string",
+//         "enum": [ "readMemory" ]
+//       },
+//       "arguments": {
+//         "$ref": "#/definitions/ReadMemoryArguments"
+//       }
+//     },
+//     "required": [ "command", "arguments" ]
+//   }]
+// },
+// "ReadMemoryArguments": {
+//   "type": "object",
+//   "description": "Arguments for `readMemory` request.",
+//   "properties": {
+//     "memoryReference": {
```

Comment:
If you look at the `memory read` command it takes start and end `address-expression`. If you limit this API to memoryReference it might be hard to extend in the future.

Reply:
Note that this API here is not defined by me, but by the Debug Adapter Protocol. See https://microsoft.github.io/debug-adapter-protocol/specification#Requests_ReadMemory for the upstream definition of how the `readMemory` request should look like. Those comments here are taken from the corresponding [JSON-schema definition](https://github.com/microsoft/debug-adapter-protocol/blob/73b9891bb8398eaec943cbc54470999b2b6eb64e/debugAdapterProtocol.json#L2930)